lib/model/flag.dart

```diff
@@ -0,0 +1,103 @@
+import 'package:flagship/flagship.dart';
+import 'package:flagship/model/modification.dart';
+import 'package:flagship/utils/logger/log_manager.dart';
+import 'package:flagship/visitor/visitor_delegate.dart';
+
+class Flag<T> {
+  // Key associated to the Flag
+  final String _key;
+  // Value for the Flag
+  final T _defaultValue;
+  // Delegate
+  final VisitorDelegate _visitorDelegate;
+
+  Flag(this._key, this._defaultValue, this._visitorDelegate);
+
+// Get value for flag
+//
+// userExposed is true by default
+  T value({bool userExposed: true}) {
+    Modification? modif = this._visitorDelegate.getFlagModification(this._key);
+    if (modif != null) {
+      if (_isSameType(modif.value)) {
+        // Activate if necessary
+        if (userExposed) {
+          this.userExposed();
+        }
+        return modif.value as T;
+      }
+    }
+    return _defaultValue;
+  }
+
+// Expose Flag
+  Future<void> userExposed() async {
+    // Before expose whe should check the Type
+    Modification? modif = this._visitorDelegate.getFlagModification(this._key);
+    if (modif != null) {
+      if (modif.value == null || _isSameType(modif.value)) {
```

Comment:
Suggestion: maybe include the `modif.value == null ||` condition inside the `_isSameType` fonction ? 
By the way, in line 22, this `modif.value == null` is not checked, is it normal ?